scripts/migrate-yaml.py

```diff
@@ -0,0 +1,164 @@
+
+"""
+Print out values that have changed, by default just additions and changes.
+
+Additions can be taken as is and placed into an overrides file, but should you should verify you still want to keep it.
+
+Removals are not printed by default because not having an override is the same as taking a default.
+
+Changes are printed, showing what you overrode and what is getting overridden.
+
+CAUTION: danger-mode is only if you KNOW what you are overriding doesn't need to change, warrantee is void if used.
+"""
+
+from __future__ import print_function
+import argparse
+from deepdiff import DeepDiff
+import errno
+import yaml
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--overrides', dest='overrides', required=True,
+                        help='The path to the file with overrides.')
+    parser.add_argument('--info', dest='info', type=bool, default=False,
+                        help='Shows what is NOT overridden.')
+    parser.add_argument('--danger-mode', dest='danger', type=bool, default=False,
+                        help='Print output that is directly dumpable into an overrides file.')
+    return parser.parse_args()
+
+
+def root_cleanup(var):
+    """Cleans up what deepdiff returns so we can have some cleaner output.
+
+    :param var: the key you wish to clean up, just a string
+    :return: the cleaned up string
+    """
+    prefix = 'root[\''
+    suffix = '\']'
+    var = var[var.startswith(prefix) and len(prefix):]
+    var = var[:var.endswith(suffix) and -len(suffix)]
+    var = var.replace('\'][\'', '.')
+    return var
+
+
+def get_config(path):
+    """Load yaml data if you can
+
+    :param path: path of yaml data to load
+    :return: dictionary
+    """
+    try:
+        with open(path) as f:
+            data = f.read()
+    except IOError as e:
+        if e.errno == errno.ENOENT:
+            data = None
+        else:
+            raise e
+    else:
+        data = yaml.safe_load(data)
+
+    if data is None:
+        return {}
+    else:
+        return data
+
+
+def do_the_diff(defaults, overrides):
+    """Diff two yaml files, returning four dictionaries containing cleaned up key-value sets.
+
+    :param defaults: File you wish to compare against, the base.
+    :param overrides: File you wish to compare, the mask
+    :return: four dictionaries, what is added, what was taken away and what was changed.
+    """
+    added = {}
+    removed = {}
+    changed = {}
+    changed_types = {}
+
+    diff = DeepDiff(defaults, overrides, verbose_level=2)
+
+    if 'dictionary_item_added' in diff.keys():
+        for old_key in diff['dictionary_item_added'].keys():
+            added[root_cleanup(old_key)] = diff['dictionary_item_added'].pop(old_key)
+    if 'dictionary_item_removed' in diff.keys():
+        for old_key in diff['dictionary_item_removed'].keys():
+            removed[root_cleanup(old_key)] = diff['dictionary_item_removed'].pop(old_key)
```

Comment:
This whole section has lines that are too long and I don't have a good way of shortening it.  If someone has suggestions...